main.go

```diff
@@ -33,19 +48,56 @@ func main() {
 		_ = resp.Body.Close()
 	}()
 
+	// Check that the domain specified by the user exists in GoDaddy
+	domainExists := 0
 	for _, zone := range zones {
-		if zone.Status == "ACTIVE" {
-			log.Println(zone.Domain)
+		if strings.EqualFold(zone.Status, "active") && strings.EqualFold(zone.Domain, *userDomainName) {
+			domainExists = 1
+			log.Println("Domain", zone.Domain, "matches user domain", *userDomainName)
 		}
 	}
 
-	records, _, err := apiClient.V1Api.RecordGet(ctx, "lwts.org", "", "", nil)
+	if domainExists != 1 {
+		log.Println("Domain", *userDomainName, "does not match any domains returned from the API")
+		os.Exit(1)
+	}
+
+	var recordData map[string]string
+	recordData = make(map[string]string)
+
+	records, _, err := apiClient.V1Api.RecordGet(ctx, *userDomainName, "", "", nil)
 	for _, record := range records {
-		log.Println(record.Name, "=>", record.Data)
+		if record.Name == "@" {
```

Comment:
re-write this as:
```golang
if record.Name != "@" || record.Type_ != "A" {
    continue
}
```

Again, keep good code to the left. If you are only interested in a specific thing, check for it and return/break/continue if you don't have it. Then continue down the left path.